File: 백준/Gold/2133. 타일 채우기/타일 채우기.py
import sys
input = sys.stdin.readline

n = int(input())
dp= [0]*(n+1)

if(n %2): print(0)
else:
  dp[2] = 3
  
  for i in range(4,n+1):
    dp[i] = dp[i-2]*3 +2

    for j in range(i-4,0,-2):
      dp[i] += dp[j]*2
  
  print(dp[n])

File: 백준/Gold/2133. 타일 채우기/타일 채우기.js
const input = require("fs")
  .readFileSync("/dev/stdin", "utf-8")
  .trim()
  .split("\n");
let N = Number(input[0]);

let dp = [];
for (let i = 0; i <= 30; i++) {
  dp[i] = 0;
}
dp[0] = 1;
dp[2] = 3;


for (let i = 4; i <= N; i += 2) {
    dp[i] = dp[i-2] * dp[2];
  for (let j = i-4; j >=0; j -= 2) {
    dp[i] += dp[j]*2;
  }
}
console.log(dp[N]);